armi/reactor/blueprints/componentBlueprint.py

```diff
@@ -225,6 +225,9 @@ class can then use to modify the isotopics as necessary.
             constructedObject = components.factory(shape, [], kwargs)
             _setComponentFlags(constructedObject, self.flags, blueprint)
             insertDepletableNuclideKeys(constructedObject, blueprint)
+            constructedObject.p.theoreticalDensityFrac = (
```

Comment:
Out of curiosity, why isn't this necessary after line 217-220 above?

Reply:
That logic is for a component group. It includes a recursive call to `componentDesign.construct` to build the nested component

https://github.com/terrapower/armi/blob/f0b1accb58243001b83d7f9f4c619c9dfcc84c15/armi/reactor/blueprints/componentBlueprint.py#L212-L222

So, in theory, that inner component and material could have its theoretical density modified. I'm also not sure if component groups are used in the wild. It also looks like material modifications do not work for component groups because they pass `matMods=dict()`

https://github.com/terrapower/armi/blob/f0b1accb58243001b83d7f9f4c619c9dfcc84c15/armi/reactor/blueprints/componentBlueprint.py#L217